clang/test/CodeGen/X86/register_asm.c

```diff
@@ -0,0 +1,10 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-unknown %s -o /dev/null
+// RUN: %clang_cc1 -triple x86_64-unknown-unknown -target-feature +egpr %s -o /dev/null
+
+int foo(void) {
+  register int a __asm__("ebx");
+#ifdef __EGPR__
```

Comment:
I believe this test can be simplified and merged into an existing test.

Reply:
which one? I can only found relate one: clang/test/CodeGen/2004-03-16-AsmRegisterCrash.c but with a bad location and file name.

Reply:
We have

./clang/test/CodeGen/M68k/inline-asm-gcc-regs.c
./clang/test/CodeGen/LoongArch/inline-asm-gcc-regs.c


I suggest splitting ./clang/test/CodeGen/2004-03-16-AsmRegisterCrash.c into 

```
./clang/test/CodeGen/ARM/inline-asm-gcc-regs.c
./clang/test/CodeGen/X86/inline-asm-gcc-regs.c
```
and add your test into the second one.

Reply:
what about rename this test into clang/test/CodeGen/X86/inline-asm-gcc-regs.c?

Reply:
there's no clang/test/CodeGen/ARM/

Reply:
Create one.

Reply:
but there are many clang/test/CodeGen/aarch64-*/ And I don't know which one to put it in.

Reply:
bash$ ls -d llvm/test/CodeGen/*/

llvm/test/CodeGen/ARM/ 
llvm/test/CodeGen/AArch64/

ARM use a separate dir from AArch64. So create clang/test/CodeGen/ARM/ and put test in it.

Reply:
Why just use `#ifdef __arm__` to distinguish ARM from X86?

Reply:
> bash$ ls -d llvm/test/CodeGen/*/
> 
> llvm/test/CodeGen/ARM/ llvm/test/CodeGen/AArch64/
> 
> ARM use a separate dir from AArch64. So create clang/test/CodeGen/ARM/ and put test in it.

There's no even AArch64 under clang/test/CodeGen/. ARM's clang codegen lit tests are not organized well if you look into clang/test/CodeGen/. It'd better not create a location if it won't be used in the future.